curp/src/rpc/connect.rs

```diff
@@ -220,6 +237,98 @@ impl ConnectApi for Connect {
         req.set_timeout(timeout);
         client.fetch_leader(req).await.map_err(Into::into)
     }
+
+    async fn install_snapshot(
+        &self,
+        term: u64,
+        leader_id: ServerId,
+        snapshot: Snapshot,
+    ) -> Result<tonic::Response<InstallSnapshotResponse>, ProposeError> {
+        self.filter()?;
+
+        let mut client = self.get().await?;
+        client
+            .install_snapshot(Request::new(install_snapshot_stream(
+                term, leader_id, snapshot,
+            )))
+            .await
+            .map_err(Into::into)
+    }
+}
+
+/// Generate install snapshot stream
+fn install_snapshot_stream(
+    term: u64,
+    leader_id: ServerId,
+    snapshot: Snapshot,
+) -> impl Stream<Item = InstallSnapshotRequest> {
+    // FIXME: The following code is better. But it will result in an unknown compiling error that might origin from a compiler bug(https://github.com/rust-lang/rust/issues/102211).
```

Comment:
I recommend leaving an issue to track this problem and putting the issue number in the comment here.

Reply:
Sure. Refer to #230.